cpp/tests/unit_tests/kernels/routing/routingTest.h

```diff
@@ -0,0 +1,447 @@
+
+template <typename T>
+constexpr void initData(T* data, int num, int rngSeed)
+{
+    std::default_random_engine rng(rngSeed);
+    std::normal_distribution dist{0.F, 1.F};
+    for (int ii = 0; ii < num; ++ii)
+    {
+        data[ii] = static_cast<T>(dist(rng));
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+inline void printSet(std::set<int32_t> const& s)
+{
+    for (auto it = s.begin(); it != s.end(); ++it)
+        TLLM_LOG_DEBUG("%d%s ", *it, std::distance(it, s.end()) == 1 ? "" : ", ");
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+inline bool checkSetEqual(int ie, std::set<int32_t> const& exp, std::set<int32_t> const& test, std::string const& name)
+{
+    bool result = true;
+    if (exp != test)
+    {
+        TLLM_LOG_DEBUG("Expert %d expected %s:\t", ie, name.c_str());
+        printSet(exp);
+        TLLM_LOG_DEBUG("\t but got:\t");
+        printSet(test);
+        TLLM_LOG_DEBUG("\n");
+        result = false;
+    }
+    return result;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename T>
+inline bool assertEqual(T* hostData, T* devData, int n, std::string const& name)
+{
+    TLLM_LOG_DEBUG("assertEqual %s", name.c_str());
+    T* hostTest = (T*) std::malloc(n * sizeof(T));
+    TLLM_CUDA_CHECK(cudaMemcpy(hostTest, devData, n * sizeof(T), cudaMemcpyDeviceToHost));
+    bool result = true;
+    int count = 0;
+    for (int ii = 0; ii < n; ++ii)
+    {
+        EXPECT_EQ(hostTest[ii], hostData[ii]);
+        if (hostTest[ii] != hostData[ii])
+        {
+            TLLM_LOG_ERROR("Unexpected result for '%s' at idx %d: exp %d != %d\n", name.c_str(), ii, (int) hostData[ii],
+                (int) hostTest[ii]);
+            result = false;
+            count++;
+
+            if (count > 10)
+            {
+                break;
+            }
+        }
+    }
+    std::free(hostTest);
+    return result;
+}
+
+template <typename T>
+inline bool printTwoVar(T* hostData, T* devData, int n, std::string const& name)
+{
+    TLLM_LOG_INFO("printTwoVar %s", name.c_str());
+    T* hostTest = (T*) std::malloc(n * sizeof(T));
+    TLLM_CUDA_CHECK(cudaMemcpy(hostTest, devData, n * sizeof(T), cudaMemcpyDeviceToHost));
+    for (int ii = 0; ii < n; ++ii)
+    {
+        std::cout << "[" << ii << "] " << hostData[ii] << " " << hostTest[ii] << " | ";
+        if (ii % 10 == 0)
+        {
+            std::cout << std::endl;
+        }
+    }
+    std::cout << std::endl;
+
+    std::free(hostTest);
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename T>
+inline bool assertClose(T* hostData, T* devData, int n, std::string const& name, float atol = 0.001, float rtol = 0.01)
```

Comment:
This function is named `assertClose`, but it does not have `assert` function. 

Is it more clear to call it `isClose` or something else?